### PR TITLE
chore(medusa): Cleanup shipping methods calculations

### PR DESCRIPTION
**Pull request by adrien2p**
**What**

Small cleanup in the tax inclusive 2 feature branch

- add few tests to validate the old tax rate usage with inclusive or not
- cleanup lint
- fix totals

### CODE REVIEW

File: packages/medusa/src/services/cart.ts [hunk 1]
```diff
@@ -1056,9 +1055,7 @@ class CartService extends TransactionBaseService {
   protected async createOrFetchUserFromEmail_(
     email: string
   ): Promise<Customer> {
-    const schema = Validator.string()
-      .email()
-      .required()
+    const schema = Validator.string().email().required()
```
Comment (olivermrbl) [inline packages/medusa/src/services/cart.ts:1058]: suggestion: We want to incrementally move away from `joi` and only use `class-validator`, so I believe we are better off using `validateEmail` from our utils. Goes for the entire file. Let me know what you think

Comment (adrien2p) [reply packages/medusa/src/services/cart.ts:1058]: yeah sure, it was not part of that PR scoped but I can still do it if you think it should be included in that pr

Comment (olivermrbl) [reply packages/medusa/src/services/cart.ts:1058]: Fine with omitting it for now. I'll create an issue for it instead 👍

Comment (adrien2p) [reply packages/medusa/src/services/cart.ts:1058]: 👍 noted

File: packages/medusa/src/strategies/tax-calculation.ts [hunk 2]
```diff
@@ -101,11 +101,13 @@ class TaxCalculationStrategy implements ITaxCalculationStrategy {
     for (const sm of shipping_methods) {
       const lineRates = taxLines.filter((tl) => tl.shipping_method_id === sm.id)
       for (const lineRate of lineRates) {
-        taxTotal += calculatePriceTaxAmount({
-          price: sm.price,
-          taxRate: lineRate.rate / 100,
-          includesTax: taxInclusiveEnabled && sm.includes_tax,
-        })
+        taxTotal += Math.round(
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/strategies/tax-calculation.ts:104]: Rounding the number inside the loop can lead to a big difference in the total.
This is a private method, and `calculate` is already rounding the total.

Comment (adrien2p) [reply packages/medusa/src/strategies/tax-calculation.ts:104]: can I let you do the changes please :)

File: packages/medusa/src/services/totals.ts [hunk 3]
```diff
@@ -210,23 +210,42 @@ class TotalsService extends TransactionBaseService {
...
+        isDefined(cartOrOrder.tax_rate) &&
+        cartOrOrder.tax_rate !== null
+      ) {
+        const taxRate = cartOrOrder.tax_rate / 100
+
+        const includesTax =
+          this.featureFlagRouter_.isFeatureEnabled(
+            TaxInclusivePricingFeatureFlag.key
+          ) && shippingMethod.includes_tax
+        const taxIncludedInPrice = !shippingMethod.includes_tax
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/services/totals.ts:224]: @srindom is there any scenario where the old `tax_rate` and tax inclusive price can coexist?

File: packages/medusa/src/services/totals.ts [hunk 4]
```diff
@@ -828,7 +848,11 @@ class TotalsService extends TransactionBaseService {
       // When we have an order with a null'ed tax rate we know that it is an
       // order from the old tax system. The following is a backward compat
       // calculation.
-      if (isOrder(cartOrOrder) && cartOrOrder.tax_rate !== null) {
+      if (
+        isOrder(cartOrOrder) &&
+        isDefined(cartOrOrder.tax_rate) &&
+        cartOrOrder.tax_rate !== null
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/services/totals.ts:854]:
```suggestion
        cartOrOrder.tax_rate != null
```

Comment (adrien2p) [reply packages/medusa/src/services/totals.ts:854]: Like it 🤘

File: packages/medusa/src/services/totals.ts (hunk 3 above)
Comment (srindom) [reply packages/medusa/src/services/totals.ts:224]: Generally, no. Could see one (super) edge case where:
- merchant has order (Order X) in their dashboard from old tax system
- merchant updates to tax-inclusive pricing version and creates a shipping option that has tax inclusive price
- merchant updates Order X with the newly created shipping option

That said, we don't have support for updating shipping options on orders through the Admin dashboard. AND the old tax system is so old that any orders from that version will (should) have been processed at this point.

I think it is okay not to address this edge case (i.e. revert the highlighted change here), but don't have strong opinions.

**Pull request closed without merging**